MyArrayList.java

```diff
@@ -1 +1,104 @@
+package com.company;
+
+public class MyArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private int position = 0;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+    private StringBuilder stringBuilder = new StringBuilder();
+    private int size = array.length;
+
+    public void printArrayList() {
+        System.out.println("My array list: ");
+        stringBuilder.setLength(0);
+        for (int j = 0; j < size; j++) {
+            stringBuilder.append(array[j]).append(", ");
+        }
+        System.out.println(stringBuilder);
+    }
+
+    @Override
+    public void add(T value) {
+        check(position);
+        add(value, position);
+        position++;
+    }
+
+    private void check(int index) {
+        if (index < 0 || index > position) {
+            if (index >= size) {
+                size += size / 2;
+                array = newArray();
+            }
+        }
+    }
+
+    private Object[] newArray() {
+        Object[] newArray = new Object[size];
+        System.arraycopy(array, 0, newArray, 0, position);
+        return newArray;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        check(index);
+        System.arraycopy(array, index, array, index + 1, size - index - 1);
+        array[index] = value;
+        size++;
+    }
+
+    @Override
+    public void addAll(List<T> list) {
+        Object[] obj = list.toArray();
+        size = array.length;
```

Comment:
do we need this line here?
size = array.length;

will this method work correct if we write:
l = new MyArrayList();
l.add(2); 
l.add(2); 
l.add(list) //  list.size() = 14;
l.size() // we should get 16